### DISCUSSION TITLE
New core team and go-forward maintenance plans

### DISCUSSION THREAD

**Discussion by domaindrivendev** [domaindrivendev/Swashbuckle.AspNetCore#2778] [Announcements]
Hi All. I'm Richie, the core maintainer of Swashbuckle - aka @domaindrivendev!

As you're probably aware, the project has suffered from a lack of meaningful maintenance over the past 12 months. This is a) because the demands of both a young family and busy day job have made it impossible for me to give it the time needed and b) because I haven't done a great job of empowering the community to carry the torch forward without me as a bottleneck.

However, despite [Microsoft's recent announcement](https://github.com/dotnet/aspnetcore/issues/54599) to remove Swashbuckle from .NET and provide similar capabilities through `Microsoft.AspNetCore.OpenApi`, I recognise that a massive number of people/teams rely heavily on Swashbuckle and are likely to do so in the short-to-medium term, while we wait to see how MS's implementation unfolds and matures. And, because of that, I wanted to take a stab at re-energizing this project. While my own time is still going to be very limited, I've been lucky enough to encounter several folks who are willing to come together as a core team and help get things back on track. So far, that team will consist of the following members, all of whom have been added as repo collaborators and therefore won't be solely dependent on me to approve and merge PRs.

- [Martin Costello](https://github.com/martincostello)
- [mesMess](https://github.com/mesMess)
- [Chris Martinez](https://github.com/commonsensesoftware)

We're currently discussing and prioritizing next steps and will post more on this shortly. But, in the mean time, here's a brief overview of what we've discussed so far:
- Goals
    - Bring project back to good health so community can continue to use it with confidence rather than seeking out alternatives
    - Provide fundamental functionality, incl. compatibility with latest versions of .NET
    - Empower broader community to contribute and bring further energy
- Non-goals
    - Compete with MS. If it makes sense for folks to use MS's implementation we're not going to try and fight that.
- Immediate next steps
    - Modernize CI/CD workflow
    - Introduce standard policies & practices for contributing going forward
    - Full support/compatability with .NET 8
    - Assess/triage/prune current list of Issues and PRs
    - Docs also need some love but that's going to be a bigger effort so will tackle above first

Please provide thoughts and feedback below as we're keen to hear from the community!

**Comment by silkfire** [comment]
Thanks for the update, really appreciated

**Comment by tonyqus** [comment]
Thank you for your efforts on this project for so many years. You did a great job!

**Comment by desjoerd** [comment]
Thanks for the update.

Question to the new maintainers: When do you plan to do a .NET 8 release?

**Reply by martincostello** [reply]
We don't have a concrete plan for the next release yet (patch vs. minor, what it will include, when it'll be published, etc.).

There's a PR already open to add .NET 8 support: #2736. If the original user picks that up again we'll take that, otherwise we (or someone else who's willing) can take a look at making a pull request to do so.

I think at first we'd just have a "targets `net8.0`" version, with _new features_ from .NET 8 coming as demand warrants after that.

We're now already building Swashbuckle with the .NET 8 SDK, so adding `net8.0` as a target (without any new features) should be relatively little work.

**Reply by desjoerd** [reply]
> We're now already building Swashbuckle with the .NET 8 SDK, so adding net8.0 as a target (without any new features) should be relatively little work.

I think that would be awesome. Just having a build with target .NET 8 (especially for the `swashbuckle.aspnetcore.cli` tool) saves us setting `DOTNET_ROLL_FORWARD=LatestMajor`.

**Comment by remcolam** [comment]
@domaindrivendev Richard, thank you for all your work before life caught up with you. And thank you for the update!

**Comment by jgarciadelanoceda** [comment]
Question for maintainers.
I am starting to search for help-wanted issues, for the issues that we do not find any problem what should we do? I have just answered an issue that from my opinion is not a issue. Should we mark the issues with labels? To start closing them

**Reply by martincostello** [reply]
Thanks for helping out.

If you've investigated an issue and it doesn't appear to be a real problem, then if you leave a comment to that effect we can untag them and wait to see if the original poster objects to the assessment.

If they don't, the stale issues automation will automatically close the issue in time.
